### ISSUE TITLE
Failed to load parser '@typescript-eslint/parser' declared in '--config » ...': globalThis is not defined

### ISSUE THREAD

**Issue by Lafaa** [eslint/eslint#14224]
**Tell us about your environment**

* **ESLint Version:** 7.10.0
* **Node Version:** 11.10.1
* **npm Version:** 6.7.0

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?** 
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "env": {
    "es6": true,
    "node": true
  },
  "globals": {
    "globalThis": "writable" 
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "sourceType": "module"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
"lint": "eslint -c .eslintrc.json --ext .ts \"src/**/*.ts\"",
```

```bash
npm run lint
```

**What did you expect to happen?**
The ESlint was working fine until yesterday we upgraded the typescript version from `3.9.3` to the `4.0.5`. This is the only thing we changed, and now the linting is not running correctly anymore. 

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
> eslint -c .eslintrc.json --ext .ts "src/**/*.ts"

(node:423) ExperimentalWarning: queueMicrotask() is experimental.

Oops! Something went wrong! :(

ESLint: 7.10.0

ReferenceError: Failed to load parser '@typescript-eslint/parser' declared in '--config » plugin:@odin/typescript-recommended » plugin:@odin/base': globalThis is not defined
    at Object.<anonymous> (/root/repo/node_modules/queue-microtask/index.js:5:25)
    at Module._compile (/root/repo/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:749:10)
    at Module.load (internal/modules/cjs/loader.js:630:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:570:12)
    at Function.Module._load (internal/modules/cjs/loader.js:562:3)
    at Module.require (internal/modules/cjs/loader.js:667:17)
    at require (/root/repo/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/root/repo/node_modules/run-parallel/index.js:4:24)
    at Module._compile (/root/repo/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
```

**Comment by mdjermanovic** [comment]
Hi @Lafaa, thanks for the issue!

> The ESlint was working fine until yesterday we upgraded the typescript version from `3.9.3` to the `4.0.5`.

It seems that `@typescript-eslint/parser` v4 has `queue-microtask` library in the dependency tree, but that library doesn't work on Node 11.10.1.

ESLint v7 and `@typescript-eslint` packages v4 don't support Node 11. Can you switch to one of the supported Node versions:

https://github.com/eslint/eslint/blob/a2013fcf996c8651bc760df21d900442828a6884/package.json#L145-L147

**Comment by Lafaa** [comment]
Hi @mdjermanovic thank you for your very fast reply! 

I will see if we can upgrade that and update the thread with the result, thank you.

**Comment by Lafaa** [comment]
Confirmed, upgrading to node 14 solved the issue. Thank you a lot!
